code/modules/projectiles/projectile.dm

```diff
@@ -182,86 +215,105 @@
-	///If TRUE, hit mobs, even if they are lying on the floor and are not our target within MAX_RANGE_HIT_PRONE_TARGETS tiles
+	/// If TRUE, hit mobs, even if they are lying on the floor and are not our target within MAX_RANGE_HIT_PRONE_TARGETS tiles
 	var/hit_prone_targets = FALSE
-	///if TRUE, ignores the range of MAX_RANGE_HIT_PRONE_TARGETS tiles of hit_prone_targets
+	/// If TRUE, ignores the range of MAX_RANGE_HIT_PRONE_TARGETS tiles of hit_prone_targets
 	var/ignore_range_hit_prone_targets = FALSE
-	///For what kind of brute wounds we're rolling for, if we're doing such a thing. Lasers obviously don't care since they do burn instead.
+	/// For what kind of brute wounds we're rolling for, if we're doing such a thing. Lasers obviously don't care since they do burn instead.
 	var/sharpness = NONE
-	///How much we want to drop damage per tile as it travels through the air
+	/// How much we want to drop damage per tile as it travels through the air
 	var/damage_falloff_tile
-	///How much we want to drop stamina damage (defined by the stamina variable) per tile as it travels through the air
+	/// How much we want to drop stamina damage (defined by the stamina variable) per tile as it travels through the air
 	var/stamina_falloff_tile
-	///How much we want to drop both wound_bonus and bare_wound_bonus (to a minimum of 0 for the latter) per tile, for falloff purposes
+	/// How much we want to drop both wound_bonus and bare_wound_bonus (to a minimum of 0 for the latter) per tile, for falloff purposes
 	var/wound_falloff_tile
-	///How much we want to drop the embed_chance value, if we can embed, per tile, for falloff purposes
+	/// How much we want to drop the embed_chance value, if we can embed, per tile, for falloff purposes
 	var/embed_falloff_tile
-	///How much accuracy is lost for each tile travelled
+	/// How much accuracy is lost for each tile travelled
 	var/accuracy_falloff = 7
-	///How much accuracy before falloff starts to matter. Formula is range - falloff * tiles travelled
+	/// How much accuracy before falloff starts to matter. Formula is range - falloff * tiles travelled
 	var/accurate_range = 100
-	var/static/list/projectile_connections = list(COMSIG_ATOM_ENTERED = PROC_REF(on_entered))
 	/// If true directly targeted turfs can be hit
 	var/can_hit_turfs = FALSE
 
 /obj/projectile/Initialize(mapload)
 	. = ..()
-	decayedRange = range
-	if(get_embed())
+	maximum_range = range
+	if (get_embed())
 		AddElement(/datum/element/embed)
-	AddElement(/datum/element/connect_loc, projectile_connections)
-
 	add_traits(list(TRAIT_FREE_HYPERSPACE_MOVEMENT, TRAIT_FREE_HYPERSPACE_SOFTCORDON_MOVEMENT), INNATE_TRAIT)
 
-/obj/projectile/proc/Range()
+/obj/projectile/Destroy()
+	if (hitscan)
+		generate_hitscan_tracers()
+	STOP_PROCESSING(SSprojectiles, src)
+	firer = null
+	original = null
+	if (movement_vector)
+		QDEL_NULL(movement_vector)
+	if (beam_points)
+		QDEL_LIST(beam_points)
+	if (last_point)
+		QDEL_NULL(last_point)
```

Comment:
IIRC, you can use the QDEL_NULL and QDEL_LIST macros without checking the provided parameters / variables. Is this some sort of micro-op?

Reply:
You're right nor was this intended to be a microop, but it does technically reduce the amount of unnecessary calls?

Reply:
Mh